org.aposin.licensescout.core/src/main/java/org/aposin/licensescout/license/ArtifactServerUtil.java

```diff
@@ -0,0 +1,174 @@
+            is.close();
+            return true;
+        } catch (IOException e) {
+            log.warn("Cannot access artifact server at: " + urlString);
+            log.warn("Result is cached, POM resolution is disabled for this run!");
+            return false;
+        }
+    }
+
+    /**
+     * @return the cachedCheckAccess
+     */
+    public final boolean isCachedCheckAccess() {
+        return cachedCheckAccess;
+    }
+
+    /**
+     * @return the nexusCentralBaseUrl
+     */
+    private final String getNexusCentralBaseUrl() {
+        return mavenCentralBaseUrl;
+    }
+
+    /**
+     * Adds licenses to an archive from information found in a Maven POM file.
+     * 
+     * @param inputStream      input source of the POM file
+     * @param archive          the archive to add the licenses to
+     * @param filePath         path of the POM file (for information only)
+     * @param licenseStoreData
+     * @param log              the logger
+     * @return true if one or more licenses have been added, false otherwise
+     */
+    public boolean addLicensesFromPom(final InputStream inputStream, final Archive archive, final String filePath,
+                                      final LicenseStoreData licenseStoreData, final ILFLog log) {
+        try {
+            log.debug("Checking POM file: " + filePath);
+            final MavenXpp3Reader xpp3Reader = new MavenXpp3Reader();
+            final Model model = xpp3Reader.read(inputStream);
+
+            final List<org.apache.maven.model.License> licenses = model.getLicenses();
+            boolean licenseFound = false;
+            for (final org.apache.maven.model.License license : licenses) {
+                final String licenseUrl = license.getUrl();
+                final String licenseName = license.getName();
+                log.debug("License name: " + licenseName);
+                log.debug("License URL: " + licenseUrl);
+                final boolean licenseFoundForUrl = LicenseUtil.handleLicenseUrl(licenseUrl, archive, filePath,
+                        licenseStoreData, log);
+                licenseFound |= licenseFoundForUrl;
+                boolean licenseFoundForName = false;
+                if (!licenseFoundForUrl) {
+                    licenseFoundForName = LicenseUtil.handleLicenseName(licenseName, archive, filePath,
+                            licenseStoreData, log);
+                    licenseFound |= licenseFoundForName;
+                }
+                if (!licenseFoundForUrl && !licenseFoundForName) {
+                    log.warn("Neither license name nor license URL mapping found for name/URL: " + licenseName + " / "
+                            + licenseUrl);
+                }
+            }
+            // try parent POM resolution only if we know we can reach the artifact server
+            if (!licenseFound && cachedCheckAccess) {
+                // try parent POM from Maven central or a proxy of it on an artifact server
```

Comment:
Move this code to a separate method that returns a boolean so to make the addLicensesFromProm method less complex.